tensorflow/python/ops/nn_ops.py

```diff
@@ -1109,7 +1109,7 @@ def dropout(x, keep_prob, noise_shape=None, seed=None, name=None):
   """
   with ops.name_scope(name, "dropout", [x]) as name:
     x = ops.convert_to_tensor(x, name="x")
-    if isinstance(keep_prob, float) and not 0 < keep_prob <= 1:
+    if (isinstance(keep_prob, int) or isinstance(keep_prob, float)) and not 0 < keep_prob <= 1:
       raise ValueError("keep_prob must be a scalar tensor or a float in the "
```

Comment:
Sorry for the drive-by, but `isinstance(keep_prob, (float, int))` would be more idiomatic (and compact!) here.